hazelcast-client/src/test/java/com/hazelcast/client/heartbeat/ClientHeartbeatTest.java

```diff
@@ -211,4 +218,66 @@ private static ClientConfig getClientConfig() {
         clientConfig.setProperty(ClientProperty.HEARTBEAT_INTERVAL.getName(), "500");
         return clientConfig;
     }
+
+
+    @Test
+    public void testAuthentication_whenHeartbeatResumed() throws Exception {
+        HazelcastInstance hazelcastInstance = hazelcastFactory.newHazelcastInstance();
+        ClientConfig config = new ClientConfig();
+        config.setProperty(ClientProperty.SHUFFLE_MEMBER_LIST.getName(), "false");
+        final HazelcastInstance client = hazelcastFactory.newHazelcastClient(config);
+        HazelcastClientInstanceImpl hazelcastClientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        final ClusterListenerSupport clientClusterService =
+                (ClusterListenerSupport) hazelcastClientInstanceImpl.getClientClusterService();
+
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
+        client.getLifecycleService().addLifecycleListener(new LifecycleListener() {
+            @Override
+            public void stateChanged(LifecycleEvent event) {
+                countDownLatch.countDown();
+            }
+        });
+
+        HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
+        blockMessagesFromInstance(instance2, client);
```

Comment:
this will cause the non-owner authentication request processed for the client but the response will not be returned. But we also want to test the case where authentication task for owner request is not run at all but it needs to be run later. can we test this case as well?